src/objectManager.cpp

```diff
@@ -195,7 +195,7 @@ VNUM ObjectManager::GetFreeRoomVnum(VNUM min, VNUM max, VNUM num)
     int i = min;
     if (num)
         {
-            return (RoomExists(num) ? num : 0);
+            return (!RoomExists(num) ? num : 0);
         }
 
     for (; i <= max; ++i)
```

Comment:
Although I didn't change it:


I believe the foundational assumption of this loop is incorrect. I think what was intended was, given a range (say 1-100), check to see if all of those in that range are available (eg, they don't exist), if ALL of them don't exist, then return the max value specified (as by that time i == max). If the calling function checks  that the return value of this function is equal to the max value passed into it, the assumption can be made that the range is available.

WHAT THIS FUNCTION ACTUALLY DOES IS: given a range (1 - 100), set i = min (i = 1), if room 1 doesn't exist, return immeiately from this function and never test the rest of the range. This function could better be written as

`    int i = min;

    while(i != max)
    {
        if(roomExists(i))
        {
            return 0; // immediately return an error, a room exists in the range, so the range is invalid
        }
    }

    if(i == max)
        return i; // i has reached max, uninterrupted, that range is available.
`